### PR TITLE
Audio metadata

### PR DESCRIPTION
**Pull request by Elad-Laufer**

### CODE REVIEW

**Comment by adavidof** [comment]
@Elad-Laufer 
1. use prettier for code style
2. for each new feature add tests
3. name the commits via conventional commits https://github.com/wix/media-platform-js-sdk#develop

**Comment by Elad-Laufer** [comment]
done

File: src/platform/management/metadata/file-metadata.ts [hunk 1]
```diff
@@ -8,17 +8,33 @@ import {
...
+} from './audio-basic-metadata';
+import {
+  AudioExtraMetadata,
+  IAudioExtraMetadata,
+} from './audio-extra-metadata';
 
 export interface IFileMetadata {
   fileDescriptor: IFileDescriptor | null;
-  basic: IImageBasicMetadata | IVideoBasicMetadata | null;
+  mediaType: string | null;
```
Comment (adavidof) [inline src/platform/management/metadata/file-metadata.ts]: `mediaType` should be enum, not string

put enum in `/src/types/media-platform/media-platform.ts` as `MediaType` enum

File: src/platform/management/metadata/file-metadata.ts [hunk 2]
```diff
@@ -47,12 +59,26 @@ export class FileMetadata implements IFileMetadata {
           );
           break;
 
+        case 'audio':
```
Comment (adavidof) [inline src/platform/management/metadata/file-metadata.ts]: `audio` change to enum `MediaType.Audio`

File: src/platform/management/metadata/file-metadata.ts [hunk 3]
```diff
@@ -47,12 +59,26 @@ export class FileMetadata implements IFileMetadata {
...
+            data.basic as IAudioBasicMetadata,
+          );
+          break;
+
+        // todo
+        // case 'font':
+        //   this.basic = new VideoBasicMetadata(
+        //       data.basic as IVideoBasicMetadata,
+        //   );
+        //   break;
```
Comment (adavidof) [inline src/platform/management/metadata/file-metadata.ts]: do we need it in current PR?

File: src/platform/management/metadata/file-metadata.ts [hunk 4]
```diff
@@ -62,5 +88,17 @@ export class FileMetadata implements IFileMetadata {
           return;
       }
     }
+
+    if (data.extra) {
+      // noinspection JSRedundantSwitchStatement
+      switch (type) {
+        case 'audio':
```
Comment (adavidof) [inline src/platform/management/metadata/file-metadata.ts]: audio change to enum MediaType.Audio

File: test/server/platform/management/audio-manager.spec.ts [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+import { AudioManager } from '../../../../src/platform/management/audio-manager';
+
+const repliesDir = __dirname + '/replies/';
+
+describe('Audio Manager', () => {
+  const configuration = new Configuration('manager.com', 'secret', 'appId');
+  const authenticator = new Authenticator(configuration);
+  const httpClient = new HTTPClient(authenticator);
+  const audioManager = new AudioManager(configuration, httpClient);
+  const sandbox = sinon.createSandbox();
```
Comment (adavidof) [inline test/server/platform/management/audio-manager.spec.ts]: redundant?

File: test/server/platform/management/audio-manager.spec.ts [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+          }),
+        );
+
+        done();
+      })
+      .catch(err => {
+        done(err);
+      });
+  });
+});
```
Comment (adavidof) [inline test/server/platform/management/audio-manager.spec.ts]: we need to test the case when we don't have the extra data
https://github.com/wix/media-platform-js-sdk/pull/50/files#diff-f09559f6cf2604c6849e9d400dd7c02cR92

File: src/platform/management/metadata/file-metadata.ts (hunk 3 above)
Comment (Elad-Laufer) [reply src/platform/management/metadata/file-metadata.ts]: It's a todo, one of many...

File: src/platform/management/metadata/file-metadata.ts [hunk 7]
```diff
@@ -27,40 +44,60 @@ export class FileMetadata implements IFileMetadata {
 
     this.fileDescriptor = new FileDescriptor(data.fileDescriptor || undefined);
 
-    if (!this.fileDescriptor.mimeType) {
-      return;
-    }
-
-    const type = this.fileDescriptor.mimeType.split('/')[0].toLowerCase();
+    const type = data.mediaType;
```
Comment (adavidof) [inline src/platform/management/metadata/file-metadata.ts]: we don't get the mediaType in requests from:
`/_api/files/ls_dir?path=%2F&pageSize=25`

Comment (Elad-Laufer) [reply src/platform/management/metadata/file-metadata.ts]: not the same response object

**Comment by adavidof** [comment]
@Elad-Laufer fix tests

**Comment by Elad-Laufer** [comment]
Please be more specific

**Comment by Elad-Laufer** [comment]
done

**Comment by davydof** [comment]
@Elad-Laufer please resolve the conflicts

**Comment by Elad-Laufer** [comment]
done

**Comment by davydof** [comment]
@Elad-Laufer `This branch cannot be rebased due to conflicts`
do:
1. pull latest master branch
2. rebase `audio-metadata` branch onto master
3. push `audio-metadata` branch to the origin

**Comment by Elad-Laufer** [comment]
You prefer rebase and merge over squash and merge?

**Pull request merged**